src/main/scala/scalismo/faces/segmentation/LoopyBPSegmentation.scala

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package scalismo.faces.segmentation
+
+import java.io.File
+
+import scalismo.faces.color._
+import scalismo.faces.image.{ImageBuffer, PixelImage}
+import scalismo.faces.gui.GUIBlock._
+import scalismo.faces.gui.{GUIFrame, ImagePanel}
+import scalismo.faces.io.PixelImageIO
+import scalismo.utils.Random
+
+import scala.collection.immutable.IndexedSeq
+
+
+// an implementation of loopy belief propagation for segmentation
```

Comment:
reference the model that is used for Loopy Belief Propagation ?

Reply:
is there more than one standard lbp-model for segmentation?

Reply:
probably not. Is there a paper that was used as a reference when implementing it?

Reply:
I'm quite sure you are wrong - loopy belief propagation is a quite clear definition of what is implemented.
However I added the publication behind LBP.